### ISSUE TITLE
Add try-state check for staking locks

### ISSUE THREAD

**Issue by gpestana** [paritytech/polkadot-sdk#3751]
Add new try-state check to staking to make ensure the staking lock are in sync with the stash's stake.

**Comment by kianenigma** [comment]
Yes, I think even with a slash happening, the two values must always be in sync.
